tests/unit/Runner/TestSuiteSorterTest.php

```diff
@@ -503,6 +503,42 @@ public function testCanSetRandomizationWithASeed(): void
         $this->assertSame($expectedOrder, $sorter->getExecutionOrder());
     }
 
+    public function testCanSetRandomizationWithDefectsFirst(): void
+    {
+        $cache = new DefaultResultCache;
+
+        $runState = [
+            'testOne'   => ['state' => TestStatus::success(), 'time' => 1],
+            'testTwo'   => ['state' => TestStatus::success(), 'time' => 1],
+            'testThree' => ['state' => TestStatus::error(), 'time' => 1],
+            'testFour'  => ['state' => TestStatus::success(), 'time' => 1],
+            'testFive'  => ['state' => TestStatus::error(), 'time' => 1],
+        ];
+
+        foreach ($runState as $testName => $data) {
+            $cache->setStatus(MultiDependencyTest::class . '::' . $testName, $data['state']);
+            $cache->setTime(MultiDependencyTest::class . '::' . $testName, $data['time']);
+        }
+
+        $sorter = new TestSuiteSorter($cache);
+
+        $suite = TestSuite::empty('test suite name');
+        $suite->addTestSuite(new ReflectionClass(MultiDependencyTest::class));
+
+        mt_srand(54321);
+        $sorter->reorderTestsInSuite($suite, TestSuiteSorter::ORDER_RANDOMIZED, false, TestSuiteSorter::ORDER_DEFECTS_FIRST);
+
+        $expectedOrder = [
+            MultiDependencyTest::class . '::testFive',
+            MultiDependencyTest::class . '::testThree',
+            MultiDependencyTest::class . '::testTwo',
+            MultiDependencyTest::class . '::testFour',
+            MultiDependencyTest::class . '::testOne',
+        ];
```

Comment:
we can see in this test the we correctly order the defect tests before the succeeding ones (as long as no dataprovider is used). this behaviour was not tested, therefore I have added the test